tests/e2e/statefulsets.go

```diff
@@ -422,6 +424,151 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-supervisor] statefulset", func
 		ssPodsAfterScaleDown = fss.GetPodList(client, statefulset)
 		gomega.Expect(len(ssPodsAfterScaleDown.Items) == int(replicas)).To(gomega.BeTrue(), "Number of Pods in the statefulset should match with number of replicas")
 	})
+
+	/*
+		verify online volume expansion on statefulset
+			1. Create a SC with allowVolumeExpansion set to 'true' in SVC
+			2. create stateful set with replica 3 using the above created SC
+			3. Once all the statefull set PODs are up follow the below step to edit stateful set
+			4. kubectl edit pvc <pvcName>  for each PVC in the StatefulSet, to increase its capacity.
+			5. kubectl delete sts --cascade=false <statefullSetName>  to delete the StatefulSet and leave its pods.
+			6. vi statefulset.yaml and edit the storage and increase the size to the size you have edited the PVC in step4
+			7. create the same stateful set again
+			8. Scaleup statefulset
+			9. Newly created stateful set should have the increased size
+			10. scale down statefulset to 0
+			11. delete statefulset and all PVC's and SC's
+	*/
+	ginkgo.It("[csi-supervisor] verify online volume expansion on statefulset", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var pvcSizeBeforeExpansion int64
+
+		ginkgo.By("Creating StorageClass for Statefulset")
+		// decide which test setup is available to run
+		profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+		scParameters[scParamStoragePolicyID] = profileID
+		// create resource quota
+		createResourceQuota(client, namespace, rqLimit, storageclassname)
+
+		//createPVCAndStorageClass(client, namespace, nil, scParameters, "", nil, "", true, "")
+		scSpec := getVSphereStorageClassSpec(storageclassname, scParameters, nil, "", "", true)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+		statefulset := GetStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		CreateStatefulSet(namespace, statefulset, client)
+		replicas := *(statefulset.Spec.Replicas)
+		// Waiting for pods status to be Ready
+		fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+		gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+		ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
+		gomega.Expect(ssPodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(), fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+		gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(), "Number of Pods in the statefulset should match with number of replicas")
```

Comment:
Test Steps does not mention scaling down the STS once it is created ?? Is the scaling down of replicas necessary here ?